scanner/e2etests/image_test.go

```diff
@@ -0,0 +1,244 @@
+package e2etests
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/google/go-containerregistry/pkg/authn"
+	"github.com/google/go-containerregistry/pkg/name"
+	v4 "github.com/stackrox/rox/generated/internalapi/scanner/v4"
+	"github.com/stackrox/rox/pkg/utils"
+	"github.com/stackrox/rox/scanner/client"
+	"github.com/stackrox/rox/scanner/indexer"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// Vulnerability describes a vulnerability in a TestCase.
+type Vulnerability struct {
+	Name        string                 `json:"Name"`
+	Description string                 `json:"Description"`
+	Link        string                 `json:"Link"`
+	Severity    string                 `json:"Severity"`
+	Metadata    map[string]interface{} `json:"Metadata"`
+	FixedBy     string                 `json:"FixedBy"`
+}
+
+// Vulnerability describes a feature in a TestCase.
+type Feature struct {
+	Name            string          `json:"Name"`
+	NamespaceName   string          `json:"NamespaceName"`
+	VersionFormat   string          `json:"VersionFormat"`
+	Version         string          `json:"Version"`
+	Vulnerabilities []Vulnerability `json:"Vulnerabilities"`
+	AddedBy         string          `json:"AddedBy"`
+	Location        string          `json:"Location"`
+	FixedBy         string          `json:"FixedBy"`
+}
+
+// TestWant describes the information wanted from the scan results in a TestCase.
+type TestWant struct {
+	Namespace string    `json:"namespace"`
+	Source    string    `json:"source"`
+	Features  []Feature `json:"expected_features"`
+}
+
+// TestArgs describes the arguments (i.e. test input) necessary to exercise the test case.
+type TestArgs struct {
+	Image    string `json:"image"`
+	Registry string `json:"registry"`
+	Username string `json:"username"`
+	Password string `json:"password"`
+}
+
+// TestCase represents on test case or scenario.
+type TestCase struct {
+	TestArgs
+	TestWant
+	// Test flags.
+	OnlyCheckSpecifiedVulns  bool `json:"only_check_specified_vulns"`
+	UncertifiedRhel          bool `json:"uncertified_rhel"`
```

Comment:
nit: `UncertifiedRHEL`

Reply:
Actually, I don't really expect us to have to use this flag anymore

Reply:
I am leaving these flags for completeness; Since I haven't looked yet at how the tests use them. We can remove later once clarified.